### PR TITLE
uart_hw_test: Fix issues in UART HW test

### PR DESCRIPTION
**Pull request by kishore-sn**
- Modify uart configuration. Before this change, uart test was failing on RTK tp1e board
- Fix issues in poll API usage
- Drain the buffers after each tx and rx operation
- Modify iperf command to run for 60 seconds
- Use kill() API instead of task_delete() API
- Modify some print statements for proper readable output

UART HW test is successful on RTK tp1x and tp1e boards.

Enable uart hw test by default in rtl8721csm/tc defconfig.

Signed-off-by: Kishore S N <kishore.sn@samsung.com>

### CODE REVIEW

**Review by sunghan-chang — APPROVED** [review]

**Comment by sunghan-chang** [comment]
Could you enable it on rtl8721csm/tc?

**Comment by kishore-sn** [comment]
> Could you enable it on rtl8721csm/tc?

Done. Verfied.

**Comment by sunghan-chang** [comment]
oops
1. could you split the commit for defconfig modification?
2. could you revert a file permission for that?

**Comment by kishore-sn** [comment]
> oops
> 
> 1. could you split the commit for defconfig modification?
> 2. could you revert a file permission for that?

Done

File: apps/examples/uart_hw_test/Kconfig [hunk 1]
```diff
@@ -4,7 +4,8 @@
 #
 
 config UART_HW_TEST
-    bool "UART HW test application"
-    default n
-        ---help---
-	Enable UART HW test application
+	bool "UART HW test application"
```
Comment (sunghan-chang) [inline apps/examples/uart_hw_test/Kconfig:7]: could you add this in another commit?

Comment (kishore-sn) [reply apps/examples/uart_hw_test/Kconfig:7]: Done

Comment (sunghan-chang) [reply apps/examples/uart_hw_test/Kconfig:7]: Oh.. I meant `uart_hw_test: Fix issues in UART HW test`. Current is also good. Thank you :)

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**